src/memory/AmdFlash.cc

```diff
@@ -758,12 +772,25 @@ bool AmdFlash::checkCommandBufferProgram()
 			const Sector& sector = getSector(cmd[2].addr);
 			if (cmd.back().value == 0x29 && getSector(cmd.back().addr) == sector) {
 				if (isWritable(sector)) {
-					// TODO de-duplicate same-address writes to the last one
-					for (auto i : xrange<size_t>(4, cmd.size() - 1)) {
-						auto ramAddr = sector.writeAddress + cmd[i].addr - sector.address;
-						uint8_t ramValue = (*ram)[ramAddr] & cmd[i].value;
-						ram->write(ramAddr, ramValue);
+					const size_t pageMask = chip.program.bufferPageSize - 1;
+					const size_t programAddress = cmd[4].addr & ~pageMask;
+					programBuffer.assign(chip.program.bufferPageSize, std::nullopt);
```

Comment:
If I understand correctly `programBuffer` will be sparse, mostly filled with `nullopt`, right?
Though the maximum value for `bufferPageSize` seems to be 256? And then iterating over a (small) sparse array (below on line 786) may still be perfectly fine.

Reply:
For normal program it is sized according to the `fastPageSize` definition, which is usually 1 but on the M29W640G it is 8. Here, single-byte programming will probably be the most common in practice, and it will usually be sparse.

For buffer program it is sized according to the `bufferPageSize` definition, which is 256 on the S29GL064S and 32 on the S29GL064N and M29W640G. Here, full buffer programming will probably be the most common, and it usually won’t be sparse.